node/src/main/scala/org/bitcoins/node/networking/peer/InventoryEsque.scala

```diff
@@ -0,0 +1,31 @@
+package org.bitcoins.node.networking.peer
+import org.bitcoins.core.p2p.Inventory
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.core.p2p.TypeIdentifier
+import org.bitcoins.core.protocol.blockchain.Block
+
+/** Type class for things that can be sent as part
+  * of a P2P inventory message
+  */
+trait InventoryEsque[T] {
+  def toInv(elem: T): Inventory
+
+}
+
+/** Default implementations of things that can be sent as
+  * a inventory message
+  */
+object InventoryEsque {
+
+  implicit val transactionInventoryEsque = new InventoryEsque[Transaction] {
```

Comment:
I don't really see the added benefit here, I think we should default to not using implicits as it makes things more complex and harder to reason about.